src/binary/Builtin.cc

```diff
@@ -632,6 +632,45 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  static int
+  tm_hash_value(VALUE hash, const char *key)
+  {
+    return NUM2INT(rb_hash_aref(hash, ID2SYM(rb_intern(key))));
+  }
+  
+  static struct tm *
+  hash_to_tm(VALUE hash)
+  {
+    static struct tm res;
```

Comment:
well, mvidner suggest to use static, which is ok, if you want to survive your pointer, but on other hand, this statis is not thread safe, so if we start using threads in yast, then we have problem if two threads call strftime method. Maybe we should consider a bit slower, but safer passing by value. or pass tm struct as reference parameter that was filled